Quick Sort

UserPOE-Frk opened the pull request:
Added quick sort algorithm (package name fixed).

Commit:
quicksort algorithm

Commit:
fixed package name

cegerard commented on quicksort.java:
```diff
@@ -0,0 +1,66 @@
+public class MyQuickSort {
```
This class must implements the ISortingAlgorithm interface

src/main/java/sortBench/algorithms/ISortingAlgorithm.java

cegerard commented on quicksort.java:
```diff
@@ -0,0 +1,66 @@
+        array[j] = temp;
+    }
+     
+    public static void main(String a[]){
```
You can get rid of this static main method.
You will never use it in this project.

cegerard commented on quicksort.java:
```diff
@@ -0,0 +1,66 @@
+public class MyQuickSort {
+     
+    private int array[];
```
private int[] array;

cegerard reviewed (commented):
add the algorithm package like in the default algorithm

`package sortBench.algorithm;`

cegerard reviewed (commented):
Put your algorithm in the right directory, like it was done with default sorting algorithm.

src/main/java/sortBench/algorithms/quickSort

cegerard commented:
Remember, you have write access on this repository so you can clone it directly and work with pull/push 
It is not needed to fork it.

Commit:
fixed java path for quicksort.java ;
fixed package name to sortBench.algorithm ;
useless main method taken off ;

ghost commented:
Got it all fixed !
(java path + package name + implementing + useless stuff)
(Just as well as mergesort.)

Should I open another request or is the pull request keeping itself up-to-date along with my new fork ?
(Sorry. Still having a better feeling with going through my GitHub fork somehow.)

cegerard commented:
@UserPOE-Frk Yes your pull request stay in sync.

Test are broken because of your class naming
To keep it simple:

1. Remove the quickSort.java file from the root of this projet (the first one you create)
2. Put your quickSort.java into src/main/java/sortBench/algorithms/quickSort path (create a new directory named quickSort)
3. In Java the class must have the same name than its file and its first letter Capitalized (CamelCase for naming). Just rename your file and your class with `QuickSort`


After that it will be good

Commit:
Fixed class name to CamelCase.
Prepared file name so as to avoid Windows issues in direct conversion into CamelCase.

Commit:
Converted name to CamelCase, getting past Windows issues.

ghost commented:
Noticed during post-tests a few hours then indeed. I've made it through that little time paradox !
Converted it all into CamelCase as well. Darn Windows messing up with it all the way...

cegerard commented:
Ok I will validate this.
You forgot to delete the quicksort.java from root directory, but you can do it in another PR if you want.

cegerard commented:
So it is OK you can merge your modifications by clicking the green button 😄
